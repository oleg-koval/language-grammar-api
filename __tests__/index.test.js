const LanguageToolClient = require('../lib/index');
const { languagesReturnValue, checkReturnValue } = require('./__responses__');

const FAKE_URL = 'https://example.org/api/v2';

const options = {
  endpoint: FAKE_URL,
};

describe('languageToolClient:', () => {
  let languageToolClient;
  beforeEach(() => {
    languageToolClient = new LanguageToolClient(options);
  });

  afterEach(() => {
    languageToolClient = null;
  });

  it('should respond with "GET /languages" response', async () => {
    languageToolClient.languages = jest.fn();
    languageToolClient.languages.mockReturnValue(languagesReturnValue.DEFAULT);

    const call = await languageToolClient.languages();

    expect(languageToolClient.languages).toBeCalledTimes(1);
    expect(call).toMatchSnapshot();
  });

  it('should respond with "POST /check" response with valid params', async () => {
    languageToolClient.check = jest.fn();
    languageToolClient.check.mockReturnValue(checkReturnValue.CORRECT_EN_US);

    const call = await languageToolClient.check({
      text: 'London is capital of Great Britain',
      language: 'en-US',
    });

    expect(languageToolClient.check).toBeCalledTimes(1);
    expect(call).toMatchSnapshot();
  });

  it('should respond with "POST /check" response with valid params, incorrect text', async () => {
    languageToolClient.check = jest.fn();
    languageToolClient.check.mockReturnValue(checkReturnValue.INCORRECT_EN_US);

    const call = await languageToolClient.check({
      text: 'dasnds dasdas sadasd',
      language: 'en-US',
    });
    expect(languageToolClient.check).toBeCalledTimes(1);
    expect(call).toMatchSnapshot();
  });

  it('should respond with "POST /check" response (400) with invalid params, missing language param', async () => {
    languageToolClient.check = jest.fn();
    languageToolClient.check.mockReturnValue(checkReturnValue.MISSING_LANGUAGE_PARAM);

    const call = await languageToolClient.check({
      text: 'dasnds dasdas sadasd',
    });
    expect(languageToolClient.check).toBeCalledTimes(1);
    expect(call).toMatchSnapshot();
  });

  it('should respond with "POST /check" response (500) with no params', async () => {
    languageToolClient.check = jest.fn();
    languageToolClient.check.mockReturnValue(checkReturnValue.PARAMS_NOT_PROVIDED);

    const call = await languageToolClient.check({});

    expect(languageToolClient.check).toBeCalledTimes(1);
    expect(call).toMatchSnapshot();
  });
});
