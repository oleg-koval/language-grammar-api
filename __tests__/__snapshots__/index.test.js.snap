// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`languageToolClient: should respond with "GET /languages" response 1`] = `
Array [
  Object {
    "code": "ast",
    "longCode": "ast-ES",
    "name": "Asturian",
  },
  Object {
    "code": "be",
    "longCode": "be-BY",
    "name": "Belarusian",
  },
  Object {
    "code": "br",
    "longCode": "br-FR",
    "name": "Breton",
  },
  Object {
    "code": "ca",
    "longCode": "ca-ES",
    "name": "Catalan",
  },
  Object {
    "code": "ca",
    "longCode": "ca-ES-valencia",
    "name": "Catalan (Valencian)",
  },
  Object {
    "code": "zh",
    "longCode": "zh-CN",
    "name": "Chinese",
  },
  Object {
    "code": "da",
    "longCode": "da-DK",
    "name": "Danish",
  },
  Object {
    "code": "nl",
    "longCode": "nl",
    "name": "Dutch",
  },
  Object {
    "code": "en",
    "longCode": "en",
    "name": "English",
  },
  Object {
    "code": "en",
    "longCode": "en-AU",
    "name": "English (Australian)",
  },
  Object {
    "code": "en",
    "longCode": "en-CA",
    "name": "English (Canadian)",
  },
  Object {
    "code": "en",
    "longCode": "en-GB",
    "name": "English (GB)",
  },
  Object {
    "code": "en",
    "longCode": "en-NZ",
    "name": "English (New Zealand)",
  },
  Object {
    "code": "en",
    "longCode": "en-ZA",
    "name": "English (South African)",
  },
  Object {
    "code": "en",
    "longCode": "en-US",
    "name": "English (US)",
  },
  Object {
    "code": "eo",
    "longCode": "eo",
    "name": "Esperanto",
  },
  Object {
    "code": "fr",
    "longCode": "fr",
    "name": "French",
  },
  Object {
    "code": "gl",
    "longCode": "gl-ES",
    "name": "Galician",
  },
  Object {
    "code": "de",
    "longCode": "de",
    "name": "German",
  },
  Object {
    "code": "de",
    "longCode": "de-AT",
    "name": "German (Austria)",
  },
  Object {
    "code": "de",
    "longCode": "de-DE",
    "name": "German (Germany)",
  },
  Object {
    "code": "de",
    "longCode": "de-CH",
    "name": "German (Swiss)",
  },
  Object {
    "code": "el",
    "longCode": "el-GR",
    "name": "Greek",
  },
  Object {
    "code": "it",
    "longCode": "it",
    "name": "Italian",
  },
  Object {
    "code": "ja",
    "longCode": "ja-JP",
    "name": "Japanese",
  },
  Object {
    "code": "km",
    "longCode": "km-KH",
    "name": "Khmer",
  },
  Object {
    "code": "fa",
    "longCode": "fa",
    "name": "Persian",
  },
  Object {
    "code": "pl",
    "longCode": "pl-PL",
    "name": "Polish",
  },
  Object {
    "code": "pt",
    "longCode": "pt",
    "name": "Portuguese",
  },
  Object {
    "code": "pt",
    "longCode": "pt-AO",
    "name": "Portuguese (Angola preAO)",
  },
  Object {
    "code": "pt",
    "longCode": "pt-BR",
    "name": "Portuguese (Brazil)",
  },
  Object {
    "code": "pt",
    "longCode": "pt-MZ",
    "name": "Portuguese (Mo√ßambique preAO)",
  },
  Object {
    "code": "pt",
    "longCode": "pt-PT",
    "name": "Portuguese (Portugal)",
  },
  Object {
    "code": "ro",
    "longCode": "ro-RO",
    "name": "Romanian",
  },
  Object {
    "code": "ru",
    "longCode": "ru-RU",
    "name": "Russian",
  },
  Object {
    "code": "de-DE-x-simple-language",
    "longCode": "de-DE-x-simple-language",
    "name": "Simple German",
  },
  Object {
    "code": "sk",
    "longCode": "sk-SK",
    "name": "Slovak",
  },
  Object {
    "code": "sl",
    "longCode": "sl-SI",
    "name": "Slovenian",
  },
  Object {
    "code": "es",
    "longCode": "es",
    "name": "Spanish",
  },
  Object {
    "code": "sv",
    "longCode": "sv",
    "name": "Swedish",
  },
  Object {
    "code": "tl",
    "longCode": "tl-PH",
    "name": "Tagalog",
  },
  Object {
    "code": "ta",
    "longCode": "ta-IN",
    "name": "Tamil",
  },
  Object {
    "code": "uk",
    "longCode": "uk-UA",
    "name": "Ukrainian",
  },
]
`;

exports[`languageToolClient: should respond with "POST /check" response (400) with invalid params, missing language param 1`] = `"Error: Missing 'language' parameter, e.g. 'language=en-US' for American English or 'language=fr' for French"`;

exports[`languageToolClient: should respond with "POST /check" response (500) with no params 1`] = `"Error: Internal Error: Missing 'text' or 'data' parameter"`;

exports[`languageToolClient: should respond with "POST /check" response with valid params 1`] = `
Object {
  "language": Object {
    "code": "en-US",
    "detectedLanguage": Object {
      "code": "en-US",
      "confidence": 0.99,
      "name": "English (US)",
    },
    "name": "English (US)",
  },
  "matches": Array [],
  "software": Object {
    "apiVersion": 1,
    "buildDate": "2019-03-08 21:02",
    "name": "LanguageTool",
    "premium": false,
    "premiumHint": "You might be missing errors only the Premium version can find. Contact us at support<at>languagetoolplus.com.",
    "status": "",
    "version": "4.5-SNAPSHOT",
  },
  "warnings": Object {
    "incompleteResults": false,
  },
}
`;

exports[`languageToolClient: should respond with "POST /check" response with valid params, incorrect text 1`] = `
Object {
  "language": Object {
    "code": "en-US",
    "detectedLanguage": Object {
      "code": "de-DE",
      "confidence": 0.66,
      "name": "German (Germany)",
    },
    "name": "English (US)",
  },
  "matches": Array [
    Object {
      "context": Object {
        "length": 6,
        "offset": 0,
        "text": "dasnds dasdas sadasd",
      },
      "contextForSureMatch": 0,
      "ignoreForIncompleteSentence": false,
      "length": 6,
      "message": "Possible spelling mistake found",
      "offset": 0,
      "replacements": Array [
        Object {
          "value": "bands",
        },
        Object {
          "value": "hands",
        },
        Object {
          "shortDescription": "plural of 'land'",
          "value": "lands",
        },
        Object {
          "value": "sands",
        },
        Object {
          "value": "Danes",
        },
        Object {
          "value": "dashes",
        },
        Object {
          "value": "Davids",
        },
        Object {
          "value": "dandy",
        },
        Object {
          "value": "wands",
        },
        Object {
          "value": "dads",
        },
        Object {
          "value": "dawns",
        },
        Object {
          "value": "deists",
        },
        Object {
          "value": "deigns",
        },
        Object {
          "value": "darns",
        },
        Object {
          "value": "daunts",
        },
        Object {
          "value": "damns",
        },
        Object {
          "value": "dangs",
        },
      ],
      "rule": Object {
        "category": Object {
          "id": "TYPOS",
          "name": "Possible Typo",
        },
        "description": "Possible spelling mistake",
        "id": "MORFOLOGIK_RULE_EN_US",
        "issueType": "misspelling",
      },
      "sentence": "dasnds dasdas sadasd",
      "shortMessage": "Spelling mistake",
      "type": Object {
        "typeName": "Other",
      },
    },
    Object {
      "context": Object {
        "length": 6,
        "offset": 7,
        "text": "dasnds dasdas sadasd",
      },
      "contextForSureMatch": 0,
      "ignoreForIncompleteSentence": false,
      "length": 6,
      "message": "Possible spelling mistake found",
      "offset": 7,
      "replacements": Array [
        Object {
          "value": "Dallas",
        },
        Object {
          "value": "NASDAQ",
        },
        Object {
          "value": "Dada",
        },
        Object {
          "value": "dashes",
        },
        Object {
          "value": "pandas",
        },
        Object {
          "value": "dads",
        },
        Object {
          "value": "dachas",
        },
        Object {
          "value": "deists",
        },
        Object {
          "value": "pashas",
        },
        Object {
          "value": "pastas",
        },
        Object {
          "value": "Adas",
        },
        Object {
          "value": "Haidas",
        },
      ],
      "rule": Object {
        "category": Object {
          "id": "TYPOS",
          "name": "Possible Typo",
        },
        "description": "Possible spelling mistake",
        "id": "MORFOLOGIK_RULE_EN_US",
        "issueType": "misspelling",
      },
      "sentence": "dasnds dasdas sadasd",
      "shortMessage": "Spelling mistake",
      "type": Object {
        "typeName": "Other",
      },
    },
    Object {
      "context": Object {
        "length": 6,
        "offset": 14,
        "text": "dasnds dasdas sadasd",
      },
      "contextForSureMatch": 0,
      "ignoreForIncompleteSentence": false,
      "length": 6,
      "message": "Possible spelling mistake found",
      "offset": 14,
      "replacements": Array [
        Object {
          "value": "salad",
        },
        Object {
          "value": "sagas",
        },
        Object {
          "value": "salads",
        },
        Object {
          "value": "Sadat",
        },
        Object {
          "value": "sadism",
        },
        Object {
          "value": "sadist",
        },
        Object {
          "value": "sodas",
        },
        Object {
          "value": "Adas",
        },
        Object {
          "value": "Salas",
        },
        Object {
          "value": "sades",
        },
      ],
      "rule": Object {
        "category": Object {
          "id": "TYPOS",
          "name": "Possible Typo",
        },
        "description": "Possible spelling mistake",
        "id": "MORFOLOGIK_RULE_EN_US",
        "issueType": "misspelling",
      },
      "sentence": "dasnds dasdas sadasd",
      "shortMessage": "Spelling mistake",
      "type": Object {
        "typeName": "Other",
      },
    },
  ],
  "software": Object {
    "apiVersion": 1,
    "buildDate": "2019-03-08 21:02",
    "name": "LanguageTool",
    "premium": false,
    "premiumHint": "You might be missing errors only the Premium version can find. Contact us at support<at>languagetoolplus.com.",
    "status": "",
    "version": "4.5-SNAPSHOT",
  },
  "warnings": Object {
    "incompleteResults": false,
  },
}
`;
